src/blob.rs

```diff
@@ -320,113 +318,121 @@ impl<'a> BlobObject<'a> {
-        }
-
-        let img_wh =
+        let (img_wh, max_bytes) =
             match MediaQuality::from_i32(context.get_config_int(Config::MediaQuality).await?)
                 .unwrap_or_default()
             {
-                MediaQuality::Balanced => BALANCED_IMAGE_SIZE,
-                MediaQuality::Worse => WORSE_IMAGE_SIZE,
+                MediaQuality::Balanced => (
+                    constants::BALANCED_IMAGE_SIZE,
+                    constants::BALANCED_IMAGE_BYTES,
+                ),
+                MediaQuality::Worse => (constants::WORSE_IMAGE_SIZE, constants::WORSE_IMAGE_BYTES),
             };
-
-        if self
-            .recode_to_size(context, blob_abs, img_wh, None)?
-            .is_some()
+        let strict_limits = false;
+        if let Some(new_name) =
+            self.recode_to_size(context, blob_abs, img_wh, max_bytes, strict_limits)?
         {
-            return Err(format_err!(
-                "Internal error: recode_to_size(..., None) shouldn't change the name of the image"
-            ));
+            self.name = new_name;
         }
         Ok(())
     }
 
+    /// If `!strict_limits`, then if `max_bytes` is exceeded, reduce the image to `img_wh` and just
+    /// proceed with the result.
     fn recode_to_size(
         &self,
         context: &Context,
         mut blob_abs: PathBuf,
         mut img_wh: u32,
-        max_bytes: Option<usize>,
+        max_bytes: usize,
+        strict_limits: bool,
     ) -> Result<Option<String>> {
         tokio::task::block_in_place(move || {
-            let mut img = image::open(&blob_abs).context("image recode failure")?;
-            let orientation = self.get_exif_orientation(context);
+            let mut img = image::open(&blob_abs).context("image decode failure")?;
+            let (nr_bytes, exif) = self.metadata()?;
+            let orientation = exif.as_ref().map(|exif| exif_orientation(exif, context));
             let mut encoded = Vec::new();
             let mut changed_name = None;
-
-            let exceeds_width = img.width() > img_wh || img.height() > img_wh;
-
-            let do_scale =
-                exceeds_width || encoded_img_exceeds_bytes(context, &img, max_bytes, &mut encoded)?;
-            let do_rotate = matches!(orientation, Ok(90) | Ok(180) | Ok(270));
-
-            if do_scale || do_rotate {
-                if do_rotate {
-                    img = match orientation {
-                        Ok(90) => img.rotate90(),
-                        Ok(180) => img.rotate180(),
-                        Ok(270) => img.rotate270(),
-                        _ => img,
-                    }
+            let is_jpeg = message::guess_msgtype_from_suffix(Path::new(&blob_abs))
+                == Some((Viewtype::Image, "image/jpeg"));
+            // We need to rewrite JPEGs with Exif to remove metadata such as location,
+            // camera model, etc.
+            let rm_exif = is_jpeg && exif.is_some();
```

Comment:
I checked that on Android and in KDE there is no Exif in PNG screenshots. And we discussed with @r10s in https://github.com/deltachat/deltachat-core-rust/issues/3956 that currently we can leave Exif in PNGs and other non-JPEGs while there's no evidence that they can carry a sensetive info. But i think removing it is also ok (because, again, screenshots likely have no one).

To be honest, i'd rather remove it. Let's vote

Reply:
I vote for also removing from PNG if it is cheap/easy to do as for jpeg and if it won't affect the PNG image

Reply:
The only problem is that now we're not able to save PNGs, we maybe need some crate for this. On the other hand, removing Exif is just removing some bytes from the container. Will try to do

Reply:
As discussed with @link2xt in https://github.com/deltachat/deltachat-core-rust/issues/4027#issuecomment-1428201087 , let's recode images to remove Exif for now. Most of PNGs, incl. screenshots, don't contain Exif. As for animated images, IDK what to do with them, i tried to save animated PNGs as PNGs, but the `image` crate we use don't preserve animation. Maybe we should send them in original size for now, the only question is how to guess if an image is animated

Reply:
I even think that for all animated images `ViewType::Gif` must be used (maybe rename it) so that this code won't run for them

Reply:
So, we can use the `image` crate to detect animated images, but this needs separate handling of PNGs and WebPs, there is nothing like `is_animated()` function, we have to decode an image using `image::codecs` stuff and check then. Ugly. And if it's animated, handle it as `ViewType::Gif` (renamed to e.g. `ViewType::Animation`). Should be done in a separate commit of course. @adbenitez, @link2xt, do u have better ideas?

Reply:
There is also `image_utils` crate having `Info::frames: u32` thing that is > 1 for animated images, but it doesn't support PNGs and WebPs, only Gifs :) And also requires decoding first.